koptional/src/test/kotlin/com/gojuno/koptional/OptionalSpec.kt

```diff
@@ -83,4 +83,25 @@ class OptionalSpec : Spek({
 
         }
     }
+
+    describe("component1") {
+
+        context("Optional<Int>.component1") {
+
+            val (result) = 42.toOptional()
+
+            it("destructures it to Int value") {
+                assertThat(result).isEqualTo(42)
+            }
+        }
+
+        context("None.component1") {
+
+            val (result) = (null as Int?).toOptional()
+
+            it("destructures it to null") {
+                assertThat(result).isNull()
+            }
+        }
+    }
```

Comment:
What about a test with unwrapping in lambda? Something like this:

```kotlin
listOf(Some(42)).forEach { (value) ->
    assertThat(value).isEqualTo(42)
}
```

Reply:
And Kotlin compiler doesn’t have issues, yep.

Reply:
It can serve as a _checking sample_ as well, since it is a more desirable usage (at least for me).